crates/indexer/src/models/token_models/v2_token_utils.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+
+use anyhow::{Context, Result};
+use aptos_api_types::deserialize_from_string;
+use bigdecimal::BigDecimal;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ObjectCore {
+    pub allow_ungated_transfer: bool,
+    #[serde(deserialize_with = "deserialize_from_string")]
+    pub guid_creation_num: BigDecimal,
+    pub owner: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum V2TokenResource {
+    ObjectCore(ObjectCore),
+}
+
+impl V2TokenResource {
+    pub fn is_resource_supported(data_type: &str) -> bool {
+        matches!(data_type, "0x1::object::ObjectCore")
```

Comment:
This is not right. This should be looking at 0x4::token_objects::Token as well. 0x1::object::ObjectCore is for all objects

Reply:
This is just for objects. we'll handle tokens separately.

Reply:
But this is a function on V2TokenResource? Shouldn't this be in something that implements Object?